### PR TITLE
Initial support for locale config, based on formats KCMS.

### PR DESCRIPTION
**Pull request by gilir**
This is an initial support for a locale config, which make possible to switch languages and formats. It's heavily based on the KDE one, adapted to lxqt-config, and save the environment variables directly in lxqt-session (issue lxde/lxqt#546).

I have currently 2 issues which I was unable to solve : 
- I have a double free crash when closing the app, and I think it's because I use 2 LXQt settings, but I don't know why the crash still exist.
- I don't know how to generate the template for the translation (if I missed the doc, please point me to it).

Let me know if there is other things to look at. Thanks in advance for the review :-)

### CODE REVIEW

**Comment by palinek** [comment]
@gilir I had a look on the crash issue. It is happening because `LxQt::ConfigDialog` is deleted twice -> by `qt` event dispatcher and when it goes out of scope (it's on `main`'s stack).
The solution is:
- remove attribute `Qt::WA_DeleteOnClose`
```diff
--- a/lxqt-config-locale/main.cpp
+++ b/lxqt-config-locale/main.cpp
@@ -44,7 +44,6 @@ int main (int argc, char **argv)
     QObject::connect(&dialog, SIGNAL(reset()), localePage, SLOT(initControls()));
     QObject::connect(&dialog, SIGNAL(save()), localePage, SLOT(saveSettings()));

-    dialog.setAttribute(Qt::WA_DeleteOnClose);
     dialog.setWindowIcon(QIcon::fromTheme("preferences-desktop-locale"));
     dialog.show();
```
- or create the dialog on heap
```diff
--- a/lxqt-config-locale/main.cpp
+++ b/lxqt-config-locale/main.cpp
@@ -35,18 +35,18 @@ int main (int argc, char **argv)
     LxQt::SingleApplication app(argc, argv);
     LxQt::Settings settings("lxqt-config-locale");
     LxQt::Settings session_settings("lxqt-session");
-    LxQt::ConfigDialog dialog(QObject::tr("LXQt Locale Configuration"), &settings);
+    LxQt::ConfigDialog* dialog =  new LxQt::ConfigDialog(QObject::tr("LXQt Locale Configuration"), &settings);
 
-    app.setActivationWindow(&dialog);
+    app.setActivationWindow(dialog);
 
-    LocaleConfig* localePage = new LocaleConfig(&settings, &session_settings, &dialog);
-    dialog.addPage(localePage, QObject::tr("Locale Settings"), QStringList() << "preferences-desktop-locale" << "preferences-desktop");
-    QObject::connect(&dialog, SIGNAL(reset()), localePage, SLOT(initControls()));
-    QObject::connect(&dialog, SIGNAL(save()), localePage, SLOT(saveSettings()));
+    LocaleConfig* localePage = new LocaleConfig(&settings, &session_settings, dialog);
+    dialog->addPage(localePage, QObject::tr("Locale Settings"), QStringList() << "preferences-desktop-locale" << "preferences-desktop");
+    QObject::connect(dialog, SIGNAL(reset()), localePage, SLOT(initControls()));
+    QObject::connect(dialog, SIGNAL(save()), localePage, SLOT(saveSettings()));
 
-    dialog.setAttribute(Qt::WA_DeleteOnClose);
-    dialog.setWindowIcon(QIcon::fromTheme("preferences-desktop-locale"));
-    dialog.show();
+    dialog->setAttribute(Qt::WA_DeleteOnClose);
+    dialog->setWindowIcon(QIcon::fromTheme("preferences-desktop-locale"));
+    dialog->show();
 
     return app.exec();
 }
```

To the PR: if this is a wanted feature, I would say it's OK to merge.

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,9 +27,10 @@ include(LXQtTranslate)
 
 add_subdirectory(src)
 add_subdirectory(liblxqt-config-cursor)
-add_subdirectory(lxqt-config-input)
-add_subdirectory(lxqt-config-file-associations)
 add_subdirectory(lxqt-config-appearance)
+add_subdirectory(lxqt-config-file-associations)
```
Comment (palinek) [inline CMakeLists.txt]: Little note: when you are reordering the lines... here it seems that you are also removing some stuff on first sight. If you want the reordering happen I would suggest to do the reordering as separate commit.

**Comment by palinek** [comment]
@jleclanche do we want this feature/application to be merged?
@gilir should I finish this (fix the crash & create initial .ts file)?

**Comment by gilir** [comment]
Thanks I'll fix the crash and the Makefile tonight. I'll be interesting to know how the initial .TS is created :-) (I'll probably need it for another PR on LXQt)

**Comment by palinek** [comment]
> I'll be interesting to know how the initial .TS is created :-) 

I'm also new in QT and was wondering how it works -> have a look here http://doc.qt.io/qt-5/internationalization.html
... actually there is a tool for it and it is simple as the following example (if `.ts` does not exist, it is created):
```
$ lupdate lxqt-config-locale -ts lxqt-config-locale/translations/lxqt-config-locale.ts
Scanning directory 'lxqt-config-locale'...
Updating 'lxqt-config-locale/translations/lxqt-config-locale.ts'...
    Found 20 source text(s) (20 new and 0 already existing)
```

**Comment by gilir** [comment]
Branch updated with the different comments

**Comment by luis-pereira** [comment]
@gilir To update the translations sources:
```cmake -DUPDATE_TRASNLATIONS=Yes .. && make```
The build system will generate/update the translations automatically.
After that disable the update translations mode by:
```cmake -DUPDATE_TRASNLATIONS=No ..```

This commit message explains it all (I think) https://github.com/lxde/liblxqt/commit/c8a513bd0845c06e7761c74403fd2bf00cd3a6b4

**Comment by paulolieuthier** [comment]
@gilir what's the status of this?

**Comment by luis-pereira** [comment]
@gilir This can be deleted, I guess.
